account_statement_import_base/models/account_journal.py

```diff
@@ -54,13 +54,11 @@ def _statement_line_import_update_unique_import_id(
         self, st_line_vals, account_number
     ):
         self.ensure_one()
-        if st_line_vals.get("unique_import_id"):
+        unique_import_id = st_line_vals.get("unique_import_id")
+        if unique_import_id:
             sanitized_acc_number = self._sanitize_bank_account_number(account_number)
             st_line_vals["unique_import_id"] = (
-                (sanitized_acc_number and sanitized_acc_number + "-" or "")
-                + str(self.id)
-                + "-"
-                + st_line_vals["unique_import_id"]
+                f"{sanitized_acc_number or ''}-{self.id}-{unique_import_id}"
```

Comment:
The original string without sanitized_acc_number will start with str(self.id) + "-", instead of "-" + str(self.id)
```suggestion
f"{sanitized_acc_number + '-' if sanitized_acc_number else ''}{self.id}-{unique_import_id}"
```

Reply:
nice catch, thanks